v2/plugin/collector.go

```diff
@@ -93,4 +93,9 @@ type CollectorDefinition interface {
 
 	// Allow metrics values not only on leaves but at any namespace level
 	AllowValuesAtAnyNamespaceLevel()
+
+	// Set global namespace prefix for all metrics (used by aggregator collector with metrics with different roots name).
+	// removePrefixFromOutput set to true removes global prefix from all namespaces when result is to be sent to agent.
+	// !! Should be set before any call to DefineMetric
```

Comment:
That should not rely on the comment like this IMO
Especially this could be easily asserted in the code - when `SetGlobalMetricPrefix()` is called you just need to check if the tree is still empty

Reply:
added
if len(cm.metricsDefinition.ListRules()) != 0 {
		return fmt.Errorf("global prefix must be set before any metric definition is provided")
	}